### PR TITLE
Fixes deprecation of log_group_name -> log_destination

### PR DESCRIPTION
**Pull request by slmingol**

### CODE REVIEW

File: main.tf [hunk 1]
```diff
@@ -42,23 +42,23 @@ resource "aws_cloudwatch_log_group" "default" {
 
 resource "aws_flow_log" "vpc" {
   count          = "${var.enabled == "true" ? 1 : 0}"
-  log_group_name = "${aws_cloudwatch_log_group.default.name}"
+  log_destination = "${aws_cloudwatch_log_group.default.arn}"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  log_destination = "${join("", aws_cloudwatch_log_group.default.*.arn)}"
```

File: main.tf [hunk 2]
```diff
@@ -42,23 +42,23 @@ resource "aws_cloudwatch_log_group" "default" {
...
   iam_role_arn   = "${aws_iam_role.log.arn}"
   vpc_id         = "${var.vpc_id}"
   traffic_type   = "${var.traffic_type}"
 }
 
 resource "aws_flow_log" "subnets" {
   count          = "${var.enabled == "true" ? length(compact(var.subnet_ids)) : 0}"
-  log_group_name = "${aws_cloudwatch_log_group.default.name}"
+  log_destination = "${aws_cloudwatch_log_group.default.arn}"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  log_destination = "${join("", aws_cloudwatch_log_group.default.*.arn)}"
```

File: main.tf [hunk 3]
```diff
@@ -42,23 +42,23 @@ resource "aws_cloudwatch_log_group" "default" {
...
   iam_role_arn   = "${aws_iam_role.log.arn}"
   subnet_id      = "${element(compact(var.subnet_ids), count.index)}"
   traffic_type   = "${var.traffic_type}"
 }
 
 resource "aws_flow_log" "eni" {
   count          = "${var.enabled == "true" ? length(compact(var.eni_ids)) : 0}"
-  log_group_name = "${aws_cloudwatch_log_group.default.name}"
+  log_destination = "${aws_cloudwatch_log_group.default.arn}"
```
Comment (aknysh) [inline main.tf]:
```suggestion
  log_destination = "${join("", aws_cloudwatch_log_group.default.*.arn)}"
```

**Comment by maximmi** [comment]
@slmingol could you address the comments, please?

**Comment by slmingol** [comment]
Need more detailed steps on what you guys want me to do. Not really following and have tried to delve into your Makefile targets and got completely lost.

**Comment by maximmi** [comment]
@slmingol may be you just forgot to commit your changes. you marked Andrew's comments as resolved, but there is no commit to address `splat+join` syntax. Leave `readme` as is, I could fix it if your repo have access to push changes

**Comment by slmingol** [comment]
@maximmi - yeah I didn't understand what Andrew was asking me to do on the outputs. I hadn't resolved the comment - https://github.com/cloudposse/terraform-aws-cloudwatch-flow-logs/pull/13#pullrequestreview-346664457 where he asked for them. Just wasn't sure what to do to satisfy what he was asking for for the splat+join and the other changes.

**Comment by slmingol** [comment]
@maximmi so one issue I'm seeing is that on MacOS some of us will have gawk installed via brew. It adds both awk/gawk to your $PATH which results in this line in the file:

```
$ cat ./build-harness/Makefile.helpers
...
...
help/generate:
	@awk '/^[a-zA-Z\_0-9%:\\\/-]+:/ { \
	  helpMessage = match(lastLine, /^## (.*)/); \`
```

That first line to `gawk/awk` is complaining about the `\_`. I uninstalled gawk and brew instaled awk to workaround that issue.

**Comment by slmingol** [comment]
@aknysh - can you explain what outputs to fix? The outputs.tf has the splat+join from what I can tell:

```
$ cat outputs.tf
output "log_group_arn" {
  value       = "${join("", aws_cloudwatch_log_group.default.*.arn)}"
  description = "ARN of the log group"
}

output "vpc_flow_id" {
  value       = "${join("", aws_flow_log.vpc.*.id)}"
  description = "VPC Flow Log ID"
}

output "subnet_flow_ids" {
  value       = "${join("", aws_flow_log.subnets.*.id)}"
  description = "Flow Log IDs of subnets"
}

output "eni_flow_ids" {
  value       = "${join("", aws_flow_log.eni.*.id)}"
  description = "Flow Log IDs of ENIs"
}

output "kinesis_id" {
  value       = "${join("", aws_kinesis_stream.default.*.id)}"
  description = "Kinesis Stream ID"
}

output "kinesis_name" {
  value       = "${join("", aws_kinesis_stream.default.*.name)}"
  description = "Kinesis Stream name"
}

output "kinesis_shard_count" {
  value       = "${join("", aws_kinesis_stream.default.*.shard_count)}"
  description = "Kinesis Stream Shard count"
}

output "kinesis_arn" {
  value       = "${join("", aws_kinesis_stream.default.*.arn)}"
  description = "Kinesis Stream ARN"
}
```

**Comment by aknysh** [comment]
@slmingol 
If all outputs have splat+join, then it's should be fine (sorry, don't remember why I asked to update them).
Please rebase (resolve conflicts)

**Comment by slmingol** [comment]
@aknysh - OK resolved conflicts.

**Comment by slmingol** [comment]
@aknysh sorry for being a noob, I now realize that you guys have Travis running at the bottom telling me what's failing, I've resolved the checks, lmk if you want all this squashed.

**Pull request merged**